concepts/websockets.textile

```diff
@@ -0,0 +1,591 @@
+
+A real solution was needed - something which would empower developers with proper TCP/IP socket-style capabilities in a web environment. Such a solution would need to be built _for the web_, and it would need to address all of the concerns that arise when operating in a web environment.
+
+h3(#enter,-websockets). Enter, WebSockets
+
+Around the middle of 2008, the pain and limitations of using Comet when implementing anything truly robust were being felt particularly keenly by developers "Michael Carter":https://en.wikipedia.org/wiki/Michael_Carter_(entrepreneur) and "Ian Hickson":https://en.wikipedia.org/wiki/Ian_Hickson. Through collaboration "on IRC":https://krijnhoetmer.nl/irc-logs/whatwg/20080618#l-1145 and "W3C mailing lists":https://lists.w3.org/Archives/Public/public-whatwg-archive/2008Jun/0165.html, they hatched a plan to introduce a new standard for modern real-time, bidirectional communication on the web, and thus "the name _"WebSocket"_ was coined":https://lists.w3.org/Archives/Public/public-whatwg-archive/2008Jun/0186.html.
+
+The idea made its way into the W3C HTML draft standard and, shortly after, Michael Carter wrote "an article introducing the Comet community to the WebSockets.":http://cometdaily.com/2008/07/04/html5-websocket/ In 2010, Google Chrome 4 was the first browser to ship full support for WebSockets, with other browser vendors following suit over the course of the next few years. In 2011, "RFC 6455 - _The WebSocket Protocol_":https://tools.ietf.org/html/rfc6455 - was published to the IETF website.
+
+Today, "all major browsers have full support for WebSockets":https://caniuse.com/#feat=websockets, even including Internet Explorer 10 and 11. In addition, browsers on both iOS and Android have supported WebSockets since 2013, which means that, all in all, the modern landscape for WebSocket support is very healthy. Much of the "internet of things" runs on some version of Android as well, so WebSocket support on other types of devices is reasonably pervasive too, as of 2018.
+
+h3(#so-what-exactly-are-websockets,-anyway?). So what exactly are WebSockets, anyway?
+
+In a nutshell, WebSockets are a thin transport layer built on top of a device's TCP/IP stack. The intent is to provide what is essentially an as-close-to-raw-as-possible TCP communication layer to web application developers, while adding a few abstractions to eliminate certain friction that would otherwise exist with respect to the way the web works. They also cater for the fact that the web has additional security considerations that must be taken into account in order to protect both consumers and service providers.
```

Comment:
Remove 'an' after 'what is essentially'

Reply:
Seeing as "as-close-to-raw-as-possible" is being used as a pseudo-adjective, and the associated noun "TCP communication layer" is singular, not plural/collective, the preceding "an" is correct.